extensions/measurement-tracking/src/viewports/TrackedCornerstoneViewport.tsx

```diff
@@ -55,6 +57,62 @@ function TrackedCornerstoneViewport(props) {
     ManufacturerModelName,
   } = displaySet.images[0];
 
+  const updateIsTracked = useCallback(() => {
+    const viewport = cornerstoneViewportService.getCornerstoneViewportByIndex(
+      viewportIndex
+    );
+
+    if (viewport instanceof BaseVolumeViewport) {
+      // A current image id will only exist for volume viewports that can have measurements tracked.
+      // Typically these are those volume viewports for the series of acquisition.
+      const currentImageId = viewport?.getCurrentImageId();
+
+      if (!currentImageId) {
+        if (isTracked) {
+          setIsTracked(false);
+        }
+        return;
+      }
+    }
+
+    if (trackedSeries.includes(SeriesInstanceUID) !== isTracked) {
+      setIsTracked(!isTracked);
+    }
+  }, [isTracked, trackedMeasurements, viewportIndex, SeriesInstanceUID]);
+
+  const onElementEnabled = useCallback(
+    evt => {
+      if (evt.detail.element !== viewportElem) {
+        setViewportElem(evt.detail.element);
+      }
+    },
+    [viewportElem]
+  );
+
+  const onElementDisabled = useCallback(evt => {
+    setViewportElem(null);
+  }, []);
+
+  useEffect(updateIsTracked, [updateIsTracked]);
+
+  /**
+   * This useEffect is needed for volume viewports so that an current image id
+   * can be reliably fetched once the volume is actually loaded.
+   */
+  useEffect(() => {
+    viewportElem?.addEventListener(
```

Comment:
can you put this addEventListener in the onElementEnabled and the clean up in the onElementDisable so that we get rid of this effect?

Reply:
Yes that would be a very good thing to do. However I tried to do but noticed a few things...

- in `OHIFCornerstoneViewport`, `onElementDisabled` is at present passing a null argument (for `viewportInfo`) because once `cornerstoneViewportService.disableElement` is called, `viewportInfo` is null (see https://github.com/OHIF/Viewers/blob/f2004aee9eae1680c0f2e4c46f102015d197c747/extensions/cornerstone/src/Viewport/OHIFCornerstoneViewport.tsx#L340)
- I tried to instead get the `viewportInfo` prior to calling `disableElement`, but then `viewportInfo.element` was null in the `onElementDisabled` callback

I checked other places that are using `onElementEnabled` and `onElementDisabled` and each of the other places uses them as they are here (see https://github.com/OHIF/Viewers/blob/f2004aee9eae1680c0f2e4c46f102015d197c747/extensions/cornerstone-dicom-rt/src/viewports/OHIFCornerstoneRTViewport.tsx#L86).

I think it might be best to leave as is for now. Thoughts?

Reply:
You are correct. I can leave the state and make this work. Thanks.